src/vue/backend/src/views/GuidedSetupConnectApi.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+    <GuidedSetupContentCard
+        cardTitle="Connect Your IDX Broker Account"
+        :steps="guidedSetupSteps"
+        :relatedLinks="links"
+        @back-step="goBackStep"
+        @skip-step="goSkipStep"
+        @continue="goContinue">
+        <template v-slot:description>
+            <p>This step is optional. A sentence or two about why you should connect IMPress for IDX Broker to your IDX Broker account. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
+        </template>
+        <template v-slot:controls>
+            <ApiKey :error="error" :loading="loading" :success="success"/>
+        </template>
+    </GuidedSetupContentCard>
+</template>
+
+<script>
+import { mapState, mapActions } from 'vuex'
+import GuidedSetupContentCard from '@/templates/GuidedSetupContentCard.vue'
+import ApiKey from '@/components/ApiKey.vue'
+export default {
+    name: 'GuidedSetupConnectApi',
+    components: {
+        ApiKey,
+        GuidedSetupContentCard
+    },
+    data () {
+        return {
+            error: false,
+            loading: false,
+            success: false,
+            links: [
+                {
+                    text: 'Where can I find my API key?',
+                    href: '#where'
+                },
+                {
+                    text: 'IDX Broker Middleware',
+                    href: 'https://middleware.idxbroker.com/mgmt/'
+                },
+                {
+                    text: 'Sign up for IDX Broker',
+                    href: '#signUp'
+                }
+            ]
+        }
+    },
+    computed: {
+        ...mapState({
+            guidedSetupSteps: state => state.general.guidedSetupSteps
+        })
+    },
+    methods: {
+        ...mapActions({
+            generalSettingsStateChange: 'general/generalSettingsStateChange',
+            promiseWithTimeout: 'general/promiseWithTimeout'
+        }),
+        async apiCall () {
+            this.loading = true
+            await this.promiseWithTimeout()
+            this.loading = false
+            this.success = true
+        },
+        goBackStep: function () {
+            // to-do: go back in history
+            this.$router.go(-1)
+        },
+        goSkipStep: function () {
+            this.$router.push({ path: '/guided-setup/connect/general' })
+        },
+        goContinue: function () {
+            this.apiCall()
+        }
+    },
+    watch: {
+        success: function () {
+            if (this.success === true) {
+                this.$router.push({ path: '/guided-setup/connect/general' })
```

Comment:
Since this only happens once, a watcher is not needed. You can take this line and move it into your goContinue function
`this.$router.push({ path: '/guided-setup/connect/general' })`

Reply:
the router was pushing before the timeout finished, moving it here solved the problem

Reply:
A watcher should be used if you have a function or change that happens repeatedly. Your issue might have been due to not waiting for the promise to resolve

Reply:
A watcher is still not needed in this context, since the action is performed once. Instead, you can do
```
async goContinue () {
            this.loading = true
// wrap this in a try catch, to catch any errors that the api call could throw
// Also, the promise with timeout is a mock api call, and should have the name of the api call that will be used in production
            await this.promiseWithTimeout()
            this.loading = false
            this.success = true
            this.cardTitle = 'Account Connected!'
            setTimeout( () => {
                 this.$router.push({ path: '/guided-setup/connect/general' })
             }, 5000)
        }
```